client/src/views/AdminView.vue

```diff
@@ -0,0 +1,468 @@
+  return modal.hasAttribute("open") && modal.getAttribute("open") != "false"
+    ? true
+    : false;
+};
+
+// Open modal
+const openModal = (modal) => {
+  if (isScrollbarVisible()) {
+    document.documentElement.style.setProperty(
+      "--scrollbar-width",
+      `${getScrollbarWidth()}px`
+    );
+  }
+  document.documentElement.classList.add(isOpenClass, openingClass);
+  setTimeout(() => {
+    visibleModal = modal;
+    document.documentElement.classList.remove(openingClass);
+  }, animationDuration);
+  modal.setAttribute("open", true);
+};
+
+// Close modal
+const closeModal = (modal) => {
+  visibleModal = null;
+  document.documentElement.classList.add(closingClass);
+  setTimeout(() => {
+    document.documentElement.classList.remove(closingClass, isOpenClass);
+    document.documentElement.style.removeProperty("--scrollbar-width");
+    modal.removeAttribute("open");
+  }, animationDuration);
+};
+
+// Close with a click outside
+document.addEventListener("click", (event) => {
+  if (visibleModal != null) {
+    const modalContent = visibleModal.querySelector("article");
+    const isClickInside = modalContent.contains(event.target);
+    !isClickInside && closeModal(visibleModal);
+  }
+});
+
+// Close with Esc key
+document.addEventListener("keydown", (event) => {
+  if (event.key === "Escape" && visibleModal != null) {
+    closeModal(visibleModal);
+  }
+});
+
+// Get scrollbar width
+const getScrollbarWidth = () => {
+  // Creating invisible container
+  const outer = document.createElement("div");
+  outer.style.visibility = "hidden";
+  outer.style.overflow = "scroll"; // forcing scrollbar to appear
+  outer.style.msOverflowStyle = "scrollbar"; // needed for WinJS apps
+  document.body.appendChild(outer);
+
+  // Creating inner element and placing it in the container
+  const inner = document.createElement("div");
+  outer.appendChild(inner);
+
+  // Calculating difference between container's full width and the child width
+  const scrollbarWidth = outer.offsetWidth - inner.offsetWidth;
+
+  // Removing temporary elements from the DOM
+  outer.parentNode.removeChild(outer);
+
+  return scrollbarWidth;
+};
+
+// Is scrollbar visible
+const isScrollbarVisible = () => {
+  return document.body.scrollHeight > screen.height;
+};
+
+// Filter data
+const filterByYear = ref(format(new Date(), "yyyy"));
+
+const dataToShow = computed(() => {
+  if (!dbData.value) return;
+  if (!filterByYear.value) return dbData.value;
+
+  return dbData.value.filter(
+    (e) => format(new Date(e.checkInDate), "yyyy") == filterByYear.value
+  );
+});
+
+// Statistics
+
+const countOccurences = (array, value) => {
+  let count = 0;
+  array.forEach((x) => {
+    if (x.landing === value) count += 1;
+  });
+  return count;
+};
+
+const statistics = computed(() => {
+  if (!dataToShow.value || dataToShow.value.length < 1) return {};
+
+  const stats = {
+    didNotStart: countOccurences(dataToShow.value, "Doch nicht gestartet"),
```

Comment:
Da das Literal "Doch nicht gestartet" in diesem Fall eigentlich als Key benutzt wird, hätte ich es zu einer Konstanten umgebaut und dann gemeinsam mit der HomeView auf diese Konstante zu gegriffen.

Gilt natürlich auch für die anderen 3.

Reply:
habs auf die todo gepackt funktioniert ja aktuell